app/uk/gov/hmrc/eoricommoncomponent/frontend/controllers/ApplicationController.scala

```diff
@@ -50,31 +50,36 @@ class ApplicationController @Inject() (
     Ok(viewStart(Journey.Register))
   }
 
+  // Below method cannot be formatted by scalafmt, so scalafmt will be disabled for it
+  // format: off
   def startSubscription(service: Service): Action[AnyContent] = ggAuthorisedUserWithEnrolmentsAction {
     implicit request => implicit loggedInUser: LoggedInUserWithEnrolments =>
-      try {
-        if (isUserEnrolledFor(loggedInUser, service)) throw SpecificEnrolmentExists(service)
-
-        loggedInUser.groupId match {
-          case Some(groupId) =>
-            hasGroupIdEnrolmentTo(groupId, service).flatMap { groupIdEnrolmentExists =>
-              if (groupIdEnrolmentExists) throw SpecificGroupIdEnrolmentExists(service)
-
-              cdsEnrolmentCheck(loggedInUser, groupId, service)
-            }
-          case None if isUserEnrolledFor(loggedInUser, CDS) =>
-            Future.successful(Redirect(routes.HasExistingEoriController.displayPage(service))) // AutoEnrolment
-          case None =>
-            Future.successful(Redirect(routes.EmailController.form(service, Journey.Subscribe))) // Whole journey
+      {
+        Future.successful(isUserEnrolledFor(loggedInUser, service)).flatMap { isUserEnrolled =>
```

Comment:
Everything need to be in the Future, because `recover` method works on future.

Previous implementation just thrown this exception which cannot be handle inside the recover block.